ipykernel/kernelbase.py

```diff
@@ -859,6 +863,41 @@ async def debug_request(self, stream, ident, parent):
                                       parent, ident)
         self.log.debug("%s", reply_msg)
 
+    # Taken from https://github.com/jupyter-server/jupyter-resource-usage/blob/e6ec53fa69fdb6de8e878974bcff006310658408/jupyter_resource_usage/metrics.py#L16
+    def get_process_metric_value(self, process, name, attribute=None):
+        try:
+            # psutil.Process methods will either return...
+            metric_value = getattr(process, name)()
+            if attribute is not None:  # ... a named tuple
+                return getattr(metric_value, attribute)
+            else:  # ... or a number
+                return metric_value
+        # Avoid littering logs with stack traces
+        # complaining about dead processes
+        except BaseException:
+            return 0
```

Comment:
Do you wan to use something "None" or do you want process metrics to always return a number ?
I don't particularly care, just wondering if you want to a way to propagate that it's not actually 0, but a no-data ?

Reply:
Going for None. Thx.